app/models/user.rb

```diff
@@ -64,7 +64,9 @@ def deleted
 
   scope :ordered_by_name, -> { order(:name) }
   scope :human_users, -> { where.not(id: [User::SYSTEM_USER_ID, User::DELETED_USER_ID]) }
-  scope :active_in_past_7_days, -> { where('current_sign_in_at > ?', 7.days.ago) }
+  scope :active_in_past_7_days, (lambda do
+    where(id: InstanceUser.unscoped.active_in_past_7_days.select('distinct user_id'))
```

Comment:
I realise you can do `where(id: InstanceUser.unscoped.active_in_past_7_days.select(:user_id).distinct)` => does that work?

Reply:
ok, will try that out!

Reply:
@weiqingtoh updated